File: src/main/scala/ge/dm/year2022/CalorieCounting.scala
package ge.dm.year2022

import ge.dm.utils.Reader

object CalorieCounting {

  def main(args: Array[String]): Unit = {
    print(solve())
  }

  def solve(): Int = {
    val data = Reader.readFileString("src/resources/year2022/Day1CalorieCounting.txt")

    var max = 0;
    var sum = 0;
    for (i <- data.indices) {
      if (data(i).isBlank) {
        if (sum > max) {
          max = sum
        }
        sum = 0;
      } else {
        sum += data(i).toInt
      }
    }
    max
  }
}

File: src/main/java/ge/dm/year2022/CalorieCounting.java
package ge.dm.year2022;

import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

public class CalorieCounting {

    public static void main(String[] args) {
        System.out.println(solve());
    }

    public static int solve() {
        List<String> lines =
                JReader.ReadFileAsStringList("src/resources/year2022/Day1CalorieCounting.txt");

        int sum=0;
        int max=0;
        for (String line : lines) {
            if (line.equals("")) {
                if (sum > max) {
                    max = sum;
                }
                sum = 0;
            } else {
                sum += Integer.parseInt(line);
            }
        }
        return max;
    }

}
